crates/bitcoin-da/src/helpers/parsers.rs

```diff
@@ -71,119 +184,315 @@ fn get_script(tx: &Transaction) -> Result<&Script, ParserError> {
+            light_client::parse_type_2_body(instructions).map(ParsedLightClientTransaction::Chunk)
+        }
+        TransactionKindLightClient::Unknown(n) => Err(ParserError::InvalidHeaderType(n)),
+    }
+}
+
+fn parse_relevant_batchproof(
+    instructions: &mut dyn Iterator<Item = Result<Instruction<'_>, ParserError>>,
+    rollup_name: &str,
+) -> Result<ParsedBatchProofTransaction, ParserError> {
+    // PushBytes(XOnlyPublicKey)
+    let _public_key = read_push_bytes(instructions)?;
+    if OP_CHECKSIGVERIFY != read_opcode(instructions)? {
+        return Err(ParserError::UnexpectedOpcode);
+    }
+
+    // Parse header
+    let header_slice = read_push_bytes(instructions)?;
+    let Some(header) = TransactionHeaderBatchProof::from_bytes(header_slice.as_bytes()) else {
+        return Err(ParserError::InvalidHeaderLength);
+    };
+
+    // Check rollup name
+    if header.rollup_name != rollup_name.as_bytes() {
+        return Err(ParserError::InvalidRollupName);
+    }
+
+    // Parse transaction body according to type
+    match header.kind {
+        TransactionKindBatchProof::SequencerCommitment => {
+            batch_proof::parse_type_0_body(instructions)
+                .map(ParsedBatchProofTransaction::SequencerCommitment)
+        }
+        TransactionKindBatchProof::Unknown(n) => Err(ParserError::InvalidHeaderType(n)),
+    }
+}
+
+fn read_instr<'a>(
+    instructions: &mut dyn Iterator<Item = Result<Instruction<'a>, ParserError>>,
+) -> Result<Instruction<'a>, ParserError> {
+    let instr = instructions
+        .next()
+        .unwrap_or(Err(ParserError::UnexpectedEndOfScript))?;
+    Ok(instr)
+}
+
+fn read_push_bytes<'a>(
+    instructions: &mut dyn Iterator<Item = Result<Instruction<'a>, ParserError>>,
+) -> Result<&'a StructPushBytes, ParserError> {
+    let instr = read_instr(instructions)?;
+    match instr {
+        PushBytes(push_bytes) => Ok(push_bytes),
+        _ => Err(ParserError::UnexpectedOpcode),
+    }
+}
+
+fn read_opcode(
+    instructions: &mut dyn Iterator<Item = Result<Instruction<'_>, ParserError>>,
+) -> Result<Opcode, ParserError> {
+    let instr = read_instr(instructions)?;
+    let Op(op) = instr else {
+        return Err(ParserError::UnexpectedOpcode);
+    };
+    Ok(op)
+}
+
+mod light_client {
+    use bitcoin::opcodes::all::{OP_DROP, OP_ENDIF, OP_IF};
+    use bitcoin::script::Instruction;
+    use bitcoin::script::Instruction::{Op, PushBytes};
+
+    use super::{
+        read_instr, read_opcode, read_push_bytes, ParsedAggregate, ParsedChunk, ParsedComplete,
+        ParserError,
+    };
+
+    // Parse transaction body of Type0
+    pub(super) fn parse_type_0_body(
+        instructions: &mut dyn Iterator<Item = Result<Instruction<'_>, ParserError>>,
+    ) -> Result<ParsedComplete, ParserError> {
+        let op_false = read_push_bytes(instructions)?;
+        if !op_false.is_empty() {
+            return Err(ParserError::UnexpectedOpcode);
+        }
```

Comment:
you don't verify if it's really OP_FALSE

Reply:
OP_FALSE is pushbytes 0 actually. I will add comments to exclude confusion